Core/include/Acts/Definitions/Direction.hpp

```diff
@@ -0,0 +1,174 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2016-2018 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+
+#include <iosfwd>
+#include <string>
+
+namespace Acts {
+
+/// The direction is always with respect to a given momentum, surface normal or
+/// other general axes
+class Direction final {
+ private:
+  enum class Value : int {
+    Negative = -1,
+    Positive = 1,
+  };
+
+ public:
+  static constexpr auto Negative = Value::Negative;
+  static constexpr auto Positive = Value::Positive;
+
+  static constexpr auto Backward = Value::Negative;
+  static constexpr auto Forward = Value::Positive;
+
+  /// This turns a signed value into a direction
+  ///
+  /// @param scalar is the signed value
+  ///
+  /// @return a direciton enum
+  static inline constexpr Direction fromScalar(ActsScalar scalar) {
+    assert(scalar != 0);
+    return scalar > 0 ? Value::Positive : Value::Negative;
+  }
+
+  /// Convert and index [0,1] to a direction e.g. for sorting in
+  /// std::array<T, 2u>
+  ///
+  /// @param index is the direction at input
+  static inline constexpr Direction fromIndex(std::size_t index) {
+    if (index == 0u) {
+      return Value::Negative;
+    }
+    return Value::Positive;
+  }
+
+  /// Convert dir to index [0,1] which allows to store direction dependent
+  /// objects in std::array<T, 2u>
+  ///
+  /// @return either 0 or 1
+  static inline constexpr std::size_t index(Direction dir) {
+    return dir.index();
+  }
+
+  /// Invert a direction
+  ///
+  /// @param dir is the direction at input
+  ///
+  /// @return an opposite direction
+  static inline constexpr Direction invert(Direction dir) {
+    return dir.invert();
+  }
```

Comment:
I prefer the object functions but there were some occasions where I could not directly call them because the literals like `Forward` are actually not of the `Direction` type but auto convertible. but maybe it is good enough explicitly convert in those cases